sound/soc/sof/loader.c

```diff
@@ -463,11 +528,34 @@ static int load_modules(struct snd_sof_dev *sdev, const struct firmware *fw)
 	return 0;
 }
 
+int strip_fw(struct snd_sof_dev *sdev, u32 bytes)
+{
+	struct snd_sof_pdata *plat_data = sdev->pdata;
+	struct firmware *stripped_fw;
+
+	stripped_fw = devm_kmalloc(sdev->dev, sizeof(*stripped_fw), GFP_KERNEL);
+
+	if (!stripped_fw)
+		return -ENOMEM;
+
+	memcpy(stripped_fw, plat_data->fw, sizeof(*stripped_fw));
+
+	stripped_fw->size -= bytes;
+	memmove((void *)&stripped_fw->data[0], &stripped_fw->data[bytes],
+		stripped_fw->size);
+	
+	kfree(plat_data->fw);
```

Comment:
that looks very suspicious. The allocation is done in request_firmware, you shouldn't muck with it.

Reply:
changed